src/Nancy/Session/CookieBasedSessions.cs

```diff
@@ -95,7 +95,7 @@ public static IObjectSerializerSelector Enable(IPipelines pipelines, Cryptograph
         {
             var cookieBasedSessionsConfiguration = new CookieBasedSessionsConfiguration(cryptographyConfiguration)
             {
-                Serializer = new DefaultObjectSerializer()
+                Serializer = new DefaultObjectSerializer(new AppDomainAssemblyCatalog())

```

Comment:
This is going to murder performance since it needs to scan the appdomain each time